django/thunderstore/frontend/api/experimental/tests/test_community_package_list.py

```diff
@@ -0,0 +1,418 @@
+
+    data = __query_api(
+        api_client, site.community.identifier, f"q={listing1.package.name}"
+    )
+
+    __assert_packages_by_listings(data, listing1)
+
+    data = __query_api(
+        api_client, site.community.identifier, f"q={listing2.package.owner.name}"
+    )
+
+    __assert_packages_by_listings(data, listing2)
+
+    data = __query_api(
+        api_client,
+        site.community.identifier,
+        f"q={listing3.package.latest.description}",
+    )
+
+    __assert_packages_by_listings(data, listing3)
+
+
+@pytest.mark.django_db
+def test_packages_are_ordered_by_update_date_by_default(api_client: APIClient) -> None:
+    site = CommunitySiteFactory()
+    listing1 = PackageListingFactory(
+        community_=site.community,
+        package_kwargs={"date_updated": "2022-02-02 01:23:45Z"},
+    )
+    listing2 = PackageListingFactory(
+        community_=site.community,
+        package_kwargs={"date_updated": "2022-02-22 01:23:45Z"},
+    )
+    listing3 = PackageListingFactory(
+        community_=site.community,
+        package_kwargs={"date_updated": "2022-02-12 01:23:45Z"},
+    )
+
+    data = __query_api(api_client, site.community.identifier)
+
+    __assert_packages_by_listings(data, [listing2, listing3, listing1])
+
+
+@pytest.mark.django_db
+def test_packages_can_be_ordered_by_creation_date(api_client: APIClient) -> None:
+    site = CommunitySiteFactory()
+    listing1 = PackageListingFactory(
+        community_=site.community,
+        package_kwargs={"date_created": "2022-02-12 01:23:45Z"},
+    )
+    listing2 = PackageListingFactory(
+        community_=site.community,
+        package_kwargs={"date_created": "2022-02-22 01:23:45Z"},
+    )
+    listing3 = PackageListingFactory(
+        community_=site.community,
+        package_kwargs={"date_created": "2022-02-02 01:23:45Z"},
+    )
+
+    data = __query_api(api_client, site.community.identifier, "ordering=newest")
+
+    __assert_packages_by_listings(data, [listing2, listing1, listing3])
+
+
+@pytest.mark.django_db
+def test_packages_can_be_ordered_by_download_count(api_client: APIClient) -> None:
+    site = CommunitySiteFactory()
+    listing1 = PackageListingFactory(
+        community_=site.community, package_version_kwargs={"downloads": 0}
+    )
+    listing2 = PackageListingFactory(
+        community_=site.community, package_version_kwargs={"downloads": 23}
+    )
+    listing3 = PackageListingFactory(
+        community_=site.community, package_version_kwargs={"downloads": 42}
+    )
+
+    data = __query_api(
+        api_client, site.community.identifier, "ordering=most-downloaded"
+    )
+
+    __assert_packages_by_listings(data, [listing3, listing2, listing1])
+
+    # Downloads of all versions are counted towards package's downloads.
+    PackageVersionFactory(
+        package=listing1.package, downloads=9001, version_number="1.0.1"
+    )
+    bust_cache_by_condition(CacheBustCondition.any_package_updated)
```

Comment:
could add an assert between new package version being created and cache being busted which ensures the cache to actually be working and stale data _is_ being returned _until_ the cache is busted.

Reply:
I feel like it should be in this test too, as otherwise it's hard to reason (via assertions) on why the cache bust is required here. Imagine if a change is made where the cache busting is no longer necessary, IMO that would warrant removing the cache bust on this test, but if it's not asserted for it won't happen.

Reply:
Fair enough.